biochemist-python/chapters/Processing_Multiple_Files_and_Writing_Files.ipynb

```diff
@@ -0,0 +1,427 @@
+    "\n",
+    "Next we are going to use a function called `glob` in the library called `glob`.  It is a little confusing since the function and the library have the same name, but we will see other examples where this is not the case later.  The output of the function `glob` is a list of all the filenames that fit the pattern specified in the input.   The input is the file location.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import glob\n",
+    "filenames = glob.glob(file_location)\n",
+    "print(filenames)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Reading multiple files with nested for loops\n",
+    "\n",
+    "Now you have a list of all the files which end in `*.pdb` in the `PDB_files` directory. To parse every file we just read in, we will use a `for` loop to go through each file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for f in filenames:\n",
+    "    outfile = open(f,'r')\n",
+    "    data = outfile.readlines()\n",
+    "    outfile.close()\n",
+    "    for line in data:\n",
+    "        if 'RESOLUTION.' in line:\n",
+    "            res_line = line\n",
+    "            words = res_line.split()\n",
+    "            resolution = float(words[3])\n",
+    "            print(resolution)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice that in this code we actually used two `for` loops, one nested inside the other.  The outer `for` loop counts over the filenames we read in earlier.  The inner `for` loop counts over the line in each file, just as we did in our previous file parsing lesson.  \n",
+    "\n",
+    "The output our code currently generates is not that useful.  It doesn't show us which file each resolution value came from.  \n",
+    "\n",
+    "We want to print the name of the molecule with the resolution. We can use `os.path.basename`, which is another function in `os.path` to get just the name of the file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "first_file = filenames[0] # look above to recall the content of filenames\n",
+    "print(first_file)\n",
+    "\n",
+    "file_name = os.path.basename(first_file)\n",
+    "print(file_name)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"exercise-this-is-a-title exercise\">\n",
+    "<p class=\"exercise-title\">Check your understanding</p>\n",
+    "    <p>How would you extract the PDB ID from the example above? </p>\n",
+    "\n",
+    "\n",
+    "    \n",
+    "```{admonition} Solution\n",
+    ":class: dropdown\n",
+    "\n",
+    "You can use the str.split function introduced in the last lesson, and split at the '.' character.\n",
+    "    \n",
```

Comment:
Need to show code block here.

Reply:
I did not know how to remove files, so I looked for instructions on [StackOverflow ](https://stackoverflow.com/questions/2047465/how-can-i-delete-a-file-from-a-git-repository/17477961#17477961)and did this.

1. On Github, lesson3a branch, I deleted the unnecessary files.
2. I went to my lesson3a branch and did a pull request (git pull main lesson3a). 

I won't do this to any other branches until I have confirmed with you that I did no damage.

Reply:
I moved the code block below this cell into the solution.